torchci/components/additionalTestInfo/TestInfo.tsx

```diff
@@ -145,6 +170,70 @@ function TDInfo({
   );
 }
 
+function RegenerateInfo({
+  workflowId,
+  runAttempt,
+  jobs,
+}: {
+  workflowId: string;
+  runAttempt: string;
+  jobs: JobData[];
+}) {
+  const session = useSession();
+  const [status, setStatus] = useState("");
+  if (session.status !== "authenticated") {
+    return <div>You must be logged in to regenerate test data</div>;
+  }
```

Comment:
This is stopping me from seeing how the regen box looks like on the preview site. Login attemps get redirected to hud.pytorch.org :P

Reply:
I hope we restrict it further to people with write access to the repo, otherwise anyone can sort of schedule those in perpetuity

Reply:
to do a workflow dispatch, you need write permissions.  Also checked that this is still true when authenticating through an oauth app with a dummy account.  Added messages about needing write perms and additional check for write permissions before trying to dispatch.